src/textual/app.py

```diff
@@ -1532,14 +1536,14 @@ async def _close_all(self) -> None:
 
         # Close all screens on the stack
         for screen in self._screen_stack:
-            if screen._running:
+            if isinstance(screen, Screen) and screen._running:
```

Comment:
Are we going to have something on the Screen stack that is not a screen?

Reply:
No, good catch. Have removed this isinstance check